clients/src/main/java/org/apache/kafka/common/telemetry/internals/MetricsEmitter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.telemetry.internals;
+
+import java.util.Collections;
+import java.util.List;
+
+import java.io.Closeable;
+
+/**
+ * An {@code MetricsEmitter} emits the values held by the {@link SinglePointMetric}, likely first converting them
```

Comment:
"likely" seems a bit odd. I would have that that the MetricsEmitter does or does not do these things.

Reply:
The reason it's likely because again depends on the implementation regarding if the emitter just keeps in-memory track of emitted metrics (which is our use case for now) and flushes them to reporter on demand where additional labels are added before exporting. But if we are directly emitting from emitter then conversion should happen within emitter implementation. However I ll re-visit this comment when adding the implementation and see if it needs to change.